component/client/chat/ChatHistory.tsx

```diff
@@ -3,111 +3,57 @@ import {ListGroup} from "react-bootstrap";
 import * as ReactDOM from "react-dom";
 
 import IData from "../service/IData";
-import RoomService from "../service/RoomService";
 import Message from "./Message";
 
-interface IHistoryState {
-  messages: JSX.Element[];
-}
-
 interface IHistoryProps {
-  username: string;
-  url: string;
-  pushMessage: string;
+  messages: IData[];
 }
 
-class ChatHistory extends React.Component<IHistoryProps, IHistoryState> {
-  private roomService: RoomService;
+class ChatHistory extends React.Component<IHistoryProps, any> {
   private historyEnd: HTMLElement;
   private shouldScroll: boolean;
   private lastPosition: number;
 
   constructor(props: any) {
     super(props);
-
-    this.state = {
-      messages: []
-    };
     this.shouldScroll = true;
-
-    this.instantiateRoomService = this.instantiateRoomService.bind(this);
-    this.createMessage = this.createMessage.bind(this);
     this.handleScroll = this.handleScroll.bind(this);
     this.setHistoryEnd = this.setHistoryEnd.bind(this);
   }
 
   public render(): JSX.Element {
+    const listGroupStyle = {
+      marginBottom: "3px",
+      marginTop: "3px"
+    };
+
+    const messages = this.props.messages.map((data, index) => (
```

Comment:
This still re-creates all message component. If we have a large amount of messages, we should try to avoid reconstructing our message components. Rather, we should just add the single new message component onto the message list.

See how it was originally done.